### PR TITLE
feat(Operations): drag horizontal vertical icons

### PR DESCRIPTION
**Pull request by Ratheeshrajan**
Contributes to #73 

**Name of icon to be animated as shown in** [Carbon Icon Library](https://carbondesignsystem.com/guidelines/icons/library/):
Drag horizontal/vertical

**Category of icon:**
operations

### CODE REVIEW

**Review by kristastarr — CHANGES_REQUESTED** [review]
The code looks good! The animations are looking really cool too! 🙌  A few comments about the animations in terms of consistency: 
1. It looks like `Horizontal vertical` has an animation pattern of thin-wide-thin and `Horizontal vertical` has a pattern of wide-thin-wide. (Hopefully that makes sense?) I like how both of them look, but I think to make it more consistent with the existing arrow icons like `Arrows horizontal` `Arrows vertical` we should maybe only have one transition, like wide-thin, instead of having that extra "bounce".  I think I like wide-thin better than thin-wide.  If this totally doesn't make sense we can meet and talk about it 😅 
2. The animation speed here looks a bit slower than `Arrows horizontal` `Arrows vertical`.  Again I think for consistency, we should try using the same animation timing and ease as those so all the arrow icons have a cohesive feel.

@johnbister @silvio-hajdin Any thoughts? 

This is a really good first pass - they are going to be an awesome addition to the library! Could you make a few tweaks based on the comments above, then we can all take a look and see if we like the changes?

File: src/components/Operations/DragHorizontal/drag-horizontal--motion.module.scss [hunk 1]
```diff
@@ -0,0 +1,44 @@
+@keyframes drag-horizontal-left {
+  0% {
+    transform: translateX(0px);
+  }
+  25% {
+    transform: translateX(1px);
+  }
+  65% {
+    transform: translateX(-1px);
+  }
```
Comment (johnbister) [inline src/components/Operations/DragHorizontal/drag-horizontal--motion.module.scss:10]:
```suggestion
  10% {
    transform: translateX(-2px);
  }
  20% {
    transform: translateX(0px);
  }
```

File: src/components/Operations/DragHorizontal/drag-horizontal--motion.module.scss [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+@keyframes drag-horizontal-right {
+  0% {
+    transform: translateX(0px);
+  }
+  25% {
+    transform: translateX(-1px);
+  }
+  65% {
+    transform: translateX(1px);
+  }
```
Comment (johnbister) [inline src/components/Operations/DragHorizontal/drag-horizontal--motion.module.scss:25]:
```suggestion
  10% {
    transform: translateX(2px);
  }
  20% {
    transform: translateX(0px);
  }
```

File: src/components/Operations/DragHorizontal/drag-horizontal--motion.module.scss [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+.DragHorizontalRight, .DragHorizontalLeft{
+  transform-origin: 50% 50%;
+  will-change: transform;
+}
+
+.isAnimating {
+  .DragHorizontalLeft {
+    animation: drag-horizontal-left 1.25s cubic-bezier(0.4, 0.14, 0.3, 1);
```
Comment (johnbister) [inline src/components/Operations/DragHorizontal/drag-horizontal--motion.module.scss:38]:
```suggestion
    animation: drag-horizontal-left 2s cubic-bezier(0.4, 0.14, 0.3, 1);
```

File: src/components/Operations/DragHorizontal/drag-horizontal--motion.module.scss [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  .DragHorizontalRight {
+    animation: drag-horizontal-right 1.25s cubic-bezier(0.4, 0.14, 0.3, 1);
```
Comment (johnbister) [inline src/components/Operations/DragHorizontal/drag-horizontal--motion.module.scss:42]:
```suggestion
    animation: drag-horizontal-right 2s cubic-bezier(0.4, 0.14, 0.3, 1);
```

File: src/components/Operations/DragVertical/drag-vertical--motion.module.scss [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+@keyframes drag-horizontal-bottom {
+  0% {
+    transform: translateY(0px);
+  }
+  25% {
+    transform: translateY(-1px);
+  }
+  65% {
+    transform: translateY(1px);
+  }
```
Comment (johnbister) [inline src/components/Operations/DragVertical/drag-vertical--motion.module.scss:25]:
```suggestion
  10% {
    transform: translateY(2px);
  }
  20% {
    transform: translateY(0px);
  }
```

File: src/components/Operations/DragVertical/drag-vertical--motion.module.scss [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+}
+
+.DragVerticalTop, .DragVerticalBottom{
+  transform-origin: 50% 50%;
+  will-change: transform;
+}
+
+.isAnimating {
+  .DragVerticalTop {
+    animation: drag-horizontal-top 1.25s cubic-bezier(0.4, 0.14, 0.3, 1);
```
Comment (johnbister) [inline src/components/Operations/DragVertical/drag-vertical--motion.module.scss:38]:
```suggestion
    animation: drag-horizontal-top 2s cubic-bezier(0.4, 0.14, 0.3, 1);
```

File: src/components/Operations/DragVertical/drag-vertical--motion.module.scss [hunk 7]
```diff
@@ -0,0 +1,44 @@
...
+  }
+
+  .DragVerticalBottom {
+    animation: drag-horizontal-bottom 1.25s cubic-bezier(0.4, 0.14, 0.3, 1);
```
Comment (johnbister) [inline src/components/Operations/DragVertical/drag-vertical--motion.module.scss:42]:
```suggestion
    animation: drag-horizontal-bottom 2s cubic-bezier(0.4, 0.14, 0.3, 1);
```

File: src/components/Operations/DragVertical/drag-vertical--motion.module.scss [hunk 8]
```diff
@@ -0,0 +1,44 @@
+@keyframes drag-horizontal-top {
+  0% {
+    transform: translateY(0px);
+  }
+  25% {
+    transform: translateY(1px);
+  }
+  65% {
+    transform: translateY(-1px);
+  }
```
Comment (johnbister) [inline src/components/Operations/DragVertical/drag-vertical--motion.module.scss:10]:
```suggestion
 10% {
    transform: translateY(-2px);
  }
 20% {
    transform: translateY(0px);
  }
```

**Comment by johnbister** [comment]
@Ratheeshrajan Hello and thanks for the contribution! Like Krista said, it looks great and just needs a few edits to make it more consistent with the larger set. I made a couple commits to speed up the pace a little and to make it so the arrows only expand once, then collapse. If you are comfortable with these changes, let us know and we can merge these icons in! Thanks again!

**Comment by Ratheeshrajan** [comment]
> 

@johnbister Thanks! it looks great and is more subtle.

**Comment by johnbister** [comment]
@Ratheeshrajan thanks for your contribution! @kristastarr if it looks good to you too, let's merge it.

**Review by johnbister — APPROVED** [review]
looks good to me! :tada:

**Review by kristastarr — APPROVED** [review]

**Comment by kristastarr** [comment]
I re-named the components from `DragHorizontal` and `DragVertical` to `DragHorizontalMotion` and `DragVerticalMotion` to be consistent with the rest of the icons. I should have caught that when I reviewed previously! Sorry @Ratheeshrajan for committing directly on your branch... I'm not really familiar with Github Desktop so probably won't make edits that way in the future 😅 Also couldn't figure out how to change the name of the `DragHorizontal` component folder in Github desktop so will do a separate PR for that rn.  But the animations look good and the code runs fine so I'll go ahead and merge this 😄

**Comment by silvio-hajdin** [comment]
Nice commit @Ratheeshrajan . This certainly can stand for now, but going further toward the v2 of it, I'd like us to look at how this icon is being used in (dynamic) UIs and components. Below is the aniGIF from one of our Carbon for IBM Product pages:

resizing_columns

Rather than implying some sort of the 'widening' or 'spreading' action coming out of it in the current release, would love to see if we can focus on animating just the pair/length of arrows (out and in), keeping the parallel lines together as they are in the static version [potentially moving them in pair as well, along the primary axis for the icon (vertical or horizontal)] . Will see if I can provider a quick rendition out of After Effects, as a reference for it. Thanks!

**Pull request merged**